source/adios2/toolkit/interop/hdf5/HDF5Common.cpp

```diff
@@ -1260,7 +1260,11 @@ void HDF5Common::ReadAttrToIO(core::IO &io)
     hsize_t numAttrs;
     // herr_t ret = H5Gget_num_objs(m_FileId, &numObj);
     H5O_info_t oinfo;
+#ifdef H5_USE_110_API
     herr_t ret = H5Oget_info(m_FileId, &oinfo);
+#else
+    herr_t ret = H5Oget_info(m_FileId, &oinfo, H5O_INFO_ALL);
+#endif
```

Comment:
Alternatively, you can adapt the code to support the new API, as you're doing here, but then you should not use this macro, and fact you don't need a new macro at all, you can just use
```cxx
#if H5_VERSION_GE(1,12,0)
  herr_t ret = H5Oget_info(m_FileId, &oinfo, H5O_INFO_ALL);
#else
  herr_t ret = H5Oget_info(m_FileId, &oinfo);
#endif
```

Reply:
yes this would also work.

Reply:
I think @germasch has the right fix here. I think HDF5's suggestion for vandalizing the CMakeLists file to do backward compatibility is pretty weird, esp. since a source code fix is required anyway.

Reply:
My point is that one can choose either way: define the compatibility macro in CMakeLists.txt, in that case one doesn't need to change the source code at all.

Or, don't define the compatibility macro, but change the source code as shown above.  That means making the source code use the 1.12 API. The previous way is telling hdf5 1.12 to still provide the 1.10-compatibile API, so the source codes doesn't need to change.

Reply:
> HDF5's suggestion for vandalizing the CMakeLists file to do backward compatibility is pretty weird

FWIW the `CMakelist.txt` change was my suggestion, not theirs.

Reply:
@chuckatkins 
Oh that clarifies things.  I will change it back to the usual H5_VERSION comparison when I merge the h5vol changes.

Reply:
@chuckatkins : You were forced into that path since you're providing a library which depends on HDF5. I think the HDF5 guys had envisioned that the workaround would be used by end users only.